addons/hr_holidays/models/hr_leave.py

```diff
@@ -359,21 +350,32 @@ def _compute_state(self):
         for leave in self:
             leave.state = 'confirm' if leave.validation_type != 'no_validation' else 'draft'
 
+    @api.depends('holiday_type', 'employee_id', 'department_id', 'mode_company_id')
+    def _compute_resource_calendar_id(self):
+        for leave in self:
+            calendar = False
+            if leave.holiday_type == 'employee':
+                calendar = leave.employee_id.resource_calendar_id
+            elif leave.holiday_type == 'department':
+                calendar = leave.department_id.company_id.resource_calendar_id
+            elif leave.holiday_type == 'company':
+                calendar = leave.mode_company_id.resource_calendar_id
+
+            leave.resource_calendar_id = calendar or self.env.company.resource_calendar_id
```

Comment:
```suggestion
            if leave.holiday_type == 'employee':
                leave.resource_calendar_id = leave.employee_id.resource_calendar_id
            elif leave.holiday_type == 'department':
                leave.resource_calendar_id = leave.department_id.company_id.resource_calendar_id
            elif leave.holiday_type == 'company':
                leave.resource_calendar_id = leave.mode_company_id.resource_calendar_id
            else:
                leave.resource_calendar_id = self.env.company.resource_calendar_id
```

Just casually browsing through to review the upgrade scripts. Still, isn't it easier to read like this? :thinking:

Reply:
Yes it is :) I just moved this piece of code around and didn't bother changing it, but might as well :)

Reply:
I see, of course you're right. Nice catch :)

Reply:
I didn't realize it before but it caused a bug so that's how I found out :)